polars/polars-core/src/chunked_array/ops/unique/rank.rs

```diff
@@ -52,12 +56,65 @@ pub(crate) fn rank(s: &Series, method: RankMethod) -> Series {
             count += 1;
         });
     }
-    let inv_ca = UInt32Chunked::new_from_aligned_vec(s.name(), inv);
 
     use RankMethod::*;
     match method {
-        Ordinal => inv_ca.into_series(),
+        Ordinal => {
+            let inv_ca = UInt32Chunked::new_from_aligned_vec(s.name(), inv);
+            inv_ca.into_series()
+        }
+        Random => {
+            // Safety:
+            // in bounds
+            let arr = unsafe { s.take_unchecked(&sort_idx_ca).unwrap() };
+            let not_consecutive_same = (&arr.slice(1, len - 1))
+                .neq(&arr.slice(0, len - 1))
+                .rechunk();
+            let obs = not_consecutive_same.downcast_iter().next().unwrap();
+
+            // Collect slice indices for sort_idx which point to ties in the original series.
+            let mut ties_indices = Vec::with_capacity(len + 1);
+            let mut ties_index: usize = 0;
+
+            ties_indices.push(ties_index);
+            obs.iter().for_each(|b| {
+                if let Some(b) = b {
+                    ties_index += 1;
+                    if b {
+                        ties_indices.push(ties_index)
+                    }
+                }
+            });
+            // Close last slice (if there where nulls in the original series, they will always be in the last slice).
+            ties_indices.push(len);
+
+            let mut sort_idx = sort_idx.to_vec();
+
+            let rng = &mut thread_rng();
+
+            // Shuffle sort_idx positions which point to ties in the original series.
+            for i in 0..(ties_indices.len() - 1) as usize {
+                let ties_index_start = ties_indices[i];
+                let ties_index_end = ties_indices[i + 1];
+                if ties_index_end - ties_index_start > 1 {
+                    let _ = &sort_idx[ties_index_start..ties_index_end].shuffle(rng);
```

Comment:
Does this return a Result? If so, let's unwrap, then we panic if someting goes wrong.

Reply:
It does not return a result:
```
error[E0599]: no method named `unwrap` found for unit type `()` in the current scope
   --> polars/polars-core/src/chunked_array/ops/unique/rank.rs:112:78
    |
112 |                     &sort_idx[ties_index_start..ties_index_end].shuffle(rng).unwrap();
    |                                                                              ^^^^^^ method not found in `()`
```

https://docs.rs/rand/0.8.4/rand/seq/trait.SliceRandom.html#tymethod.shuffle

Reply:
Ok, then I don't think we have to bind it to `let _`. It made me wonder why we did that. :P

Reply:
without `let _`: ```rust
warning: unused borrow that must be used
   --> polars/polars-core/src/chunked_array/ops/unique/rank.rs:113:21
    |
113 |                     &sort_idx[ties_index_start..ties_index_end].shuffle(rng);
    |                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the borrow produces a value
    |
    = note: `#[warn(unused_must_use)]` on by default
help: use `let _ = ...` to ignore the resulting value
    |
113 |                     let _ = &sort_idx[ties_index_start..ties_index_end].shuffle(rng);
    |                     +++++++
```

Reply:
I think we must borrow `&mut`. :thinking:

Reply:
`sort_idx[ties_index_start..ties_index_end].shuffle(rng);` seems to work.